fix: set completion_status to completed when lesson_status is passed

so-jd opened the pull request:
I found a logic bug, relating to this issue https://github.com/overhangio/openedx-scorm-xblock/issues/90 (was experiencing it as well).

In SCORM 2004 you have:
"cmi.completion_status" = "completed" / "incomplete" / "not attempted" / "unknown" - this event Indicates whether the learner has completed the SCO
"cmi.success_status" = "passed" / "failed" - Indicates whether the learner has mastered the SCO

but in SCORM 1.2:
"cmi.core.lesson_status" = “passed”/ “completed”/ “failed”/ “incomplete”/ “browsed”/ “not attempted” - Indicates whether the learner has completed AND satisfied the requirements for the SCO

In the code when evaluating "cmi.core.lesson_status" == passedwe need to also set the complete status otherwise SCROM 1.2 courses will not end successfully

ref: https://scorm.com/scorm-explained/technical-scorm/run-time/run-time-reference/#section-3

Danyal-Faheem commented:
Hi @so-jd, thank you for your contribution!

This change LGTM but can I also request you to add a changelog entry to your PR?

You can do this by installing scriv in a virtual environment using `pip install scriv` and then running `scriv create` in the root directory of this repository.

so-jd commented:
@Danyal-Faheem I updated to change log to your request, also want to bring to your attention that I made a mistake in my original branch and I fixed it, failed state is not considered complete, but passed should so I added another if statement for it

Danyal-Faheem commented:
Hi @so-jd, can I also request you to squash the commits and update the commit message and PR title to follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) standards as well as make it consistent with the changelog entry?

so-jd commented:
Hi @Danyal-Faheem I changed the PR name to `fix: set completion_status to completed when lesson_status is passed`, but what do you mean by squash the commits? in my branch? cause you can set it to squash when you merge the PR.
